polysun/main.py

```diff
@@ -74,12 +82,22 @@ def _read_config(self):
                 output_id_down = entry['output_id_down']
                 if not 0 <= shutter_id <= 240 or not 0 <= output_id_up <= 240 or not 0 <= output_id_down <= 240:
                     continue
-                self._mapping[shutter_id] = {'up': output_id_up,
-                                             'down': output_id_down}
+                new_mapping[shutter_id] = {'up': output_id_up,
+                                           'down': output_id_down}
+                for input_entry in entry.get('inputs', []):
+                    input_id = int(input_entry['input_id'])
+                    new_input_mapping.setdefault(input_id, set()).add(shutter_id)
             except ValueError:
                 self.logger('Skipped entry with shutter_id {0}'.format(shutter_id))
-        self._enabled = len(self._mapping)
+        self._mapping = new_mapping
+        self._input_shutter_mapping = new_input_mapping
+        self._enabled = len(self._mapping) > 0
         self.logger('Polysun is {0}'.format('enabled' if self._enabled else 'disabled'))
+        self._input_enabled = False
+        if len(self._input_shutter_mapping) > 0:
+            self._input_enabled = hasattr(self.webinterface, 'shutter_report_lost_position')
```

Comment:
Is there no nicer way to determine this?

Reply:
It's quite a static detection, but indeed, there might be a more elegant solution.